Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -295,9 +295,15 @@ public List<String> getRules(Game game) {
                 // evasion
                 for (Map.Entry<EvasionEffect, Set<Ability>> entry : game.getContinuousEffects().getApplicableEvasionEffects(this, game).entrySet()) {
                     for (Ability ability : entry.getValue()) {
-                        String cantBeBlockedMessage = entry.getKey().cantBeBlockedMessage(this, ability, game);
-                        if (cantBeBlockedMessage != null) {
-                            restrictHints.add(HintUtils.prepareText(cantBeBlockedMessage + addSourceObjectName(game, ability), null, HintUtils.HINT_ICON_EVASION));
+                        if (entry.getKey().hasCantBeBlockedMessage(this, ability, game)) {
```

Comment:
Why you need all that bloated code with `hasCantBeBlockedMessage`? Must use `cantBeBlocked` check to find active effect and message like:
for effect in getApplicableEvasionEffects -> if cantBeBlocked -> add message

Reply:
`cantBeBlocked` is often conditional on the blocker (like can't be blocked by green creatures).
In addition there is times it does not feel worth it to display an hint (I did not put one for ), or the logic for the cantBeBlocked ability is handled elsewhere (the CantBeBlockedByOneEffect kind), but a hint is worthwhile.

So the hint message is not correlated to a return value of `cantBeBlocked(attacker, blocker, ...)`
You commented previously that you did not like my use of `return null` as a "there is no hint message for this effect", so I did split `.cantBeBlockedMessage` in two: a boolean telling that the effect has a hint for a `cantBeBlocked` ability, and if true, a getter for the message. I guess I could use `return ""` instead of the old `return null` and only display the hint if that is not empty? I would not need the `hasCantBeBlockedMessage` method then.